lyricsgenius/artist.py

```diff
@@ -51,15 +51,15 @@ def songs(self):
     def num_songs(self):
         return self._num_songs
 
-    def add_song(self, new_song, verbose=True):
+    def add_song(self, new_song, verbose=True, include_features=False):
         """Add a Song object to the Artist object"""
 
         if any([song.title == new_song.title for song in self._songs]):
             if verbose:
                 print('{s} already in {a}, not adding song.'.format(s=new_song.title,
                                                                     a=self.name))
             return 1  # Failure
-        if new_song.artist == self.name:
+        if new_song.artist == self.name or include_features:
```

Comment:
Make this more restrictive than just allowing any artist when `include_features` is True.

Can you write an `artist_is_featured` method that examines the song title (and maybe other data?) to very that the searched artist is actually featured on the track?

Or is this extra step not necessary because Genius.com has already the featured filtering? Can you show an example of two of searching for artists and the featured tracks that get returned as part of their search?

Reply:
> Can you write an artist_is_featured method that examines the song title (and maybe other data?) to very that the searched artist is actually featured on the track?

There is a section in a song's data called "featured_artists" that I could search through to find if the current artist's name is in. However, the "featured_artists" don't always include all artists that are in a song. For example, Swae Lee is on Sicko Mode and it is currently the top song for Swae Lee in terms of page views. However, he is not credited as a featured artist on the song for whatever reason, probably because he doesn't have a lot of lines. 

> Or is this extra step not necessary because Genius.com has already the featured filtering?

Yeah Genius should be filtering through all the songs that an artist is on, even if they are technically not a featured artist.

>When I run your code the test_search_song test fails. Can you repeat this error and fix it?

How are you running the tests? I am using `coverage run -m unittest discover && coverage report` and am not replicating the failure.